axon-framework/axon-framework-commands/infrastructure.md

```diff
@@ -414,59 +376,242 @@ public DisruptorCommandBus commandBus(TransactionManager txManager, AxonConfigur
+The default load factor set for the distributed `CommandBus` implementations is 100.
+The configuration changes slightly per distributed implementation and as such will be covered in those sections.
 
-* `ERROR` - the default, and will cause an exception to be thrown when a Routing Key is not available
-* `RANDOM_KEY` - will return a random value when a \`routing key cannot be resolved from the command message.
+### Routing Strategy
 
-  This effectively means that those commands will be routed to a random segment of the command bus.
+Commands should be [routed consistently](../../architecture-overview/README.md#explicit-messaging) to the same application, especially those targeted towards a specific Aggregate.
+This ensures a single instance is in charge of the targeted aggregate, resolving the concurrent access issue and allowing for optimization like caching to work as designed.
+To component dealing with the consistent routing in an Axon application, is the `RoutingStrategy`.
 
-* `STATIC_KEY` - Will return a static key \(being "unresolved"\) for unresolved routing keys.
+The `RoutingStrategy` receives a `CommandMessage` and based on the message returns the routing key to use.
+Two commands with the same routing key will **always** be routed to the same segment, as long as there is no topology change in the distributed set-up.
 
-  This effectively means that all those commands will be routed to the same segment, as long as the configuration of segments does not change.
+At the moment, there are five implementations of the `RoutingStrategy`, where three of these are intended to be a fallback solution in case the routing key cannot be resolved:
 
-You can choose different flavor of this components that are available in one of the extension modules:
+ 1. The `AnnotationRoutingStrategy` - the **default** routing strategy expecting the `TargetAggregateIdentifier` or `RoutingKey` annotation to be present on a field inside the command class. 
+    The annotated field or getter is searched, and the contents will be returned as the routing key for that command.
+ 2. The `MetaDataRoutingStrategy` - uses a property defined during creation of this strategy to fetch the routing key from the `CommandMessage` its `MetaData`.
+ 3. The `ERROR` `UnresolvedRoutingKeyPolicy` - the **default fallback** that will cause an exception to be thrown when the routing key cannot be resolved from the given `CommandMessage`.
+ 4. The `RANDOM_KEY` `UnresolvedRoutingKeyPolicy` - will return a random value when a routing key cannot be resolved from the `CommandMessage`. 
+    This effectively means that those commands will be routed to a random segment of the command bus.
+ 5. The `STATIC_KEY` `UnresolvedRoutingKeyPolicy` - will return a static key \(named "unresolved"\) for unresolved routing keys. 
+    This effectively means that all those commands will be routed to the same segment, as long as the configuration of segments does not change.
```

Comment:
```suggestion
    This means that all those commands will be routed to the same segment, as long as the configuration of segments does not change.
```
You've used your quota for "This, effectively" in the last sentence. 😜

Reply:
Fair one! I'll reword the sentence a bit further even :-)

Reply:
FYI, changed it to this:

```
 4. The `RANDOM_KEY` `UnresolvedRoutingKeyPolicy` - will return a random value when a routing key cannot be resolved from the `CommandMessage`. 
    This means that those commands will be routed to a random segment of the command bus.
 5. The `STATIC_KEY` `UnresolvedRoutingKeyPolicy` - will return a static key \(named "unresolved"\) for unresolved routing keys. 
    This policy routes all commands to the same segment, as long as the configuration of segments does not change.
```